MSAL/src/instance/MSALDeviceInfoProvider.m

```diff
@@ -32,59 +32,91 @@
 #import "MSIDRequestParameters+Broker.h"
 #import "MSALDeviceInformation+Internal.h"
 
+#import "MSIDWorkPlaceJoinConstants.h"
+#import "MSIDWorkPlaceJoinUtil.h"
+#import "MSIDRegistrationInformation.h"
+
 @implementation MSALDeviceInfoProvider
 
 - (void)deviceInfoWithRequestParameters:(MSIDRequestParameters *)requestParameters
-                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock API_AVAILABLE(ios(13.0), macos(10.15))
+                        completionBlock:(MSALDeviceInformationCompletionBlock)completionBlock
 {
-    if (![requestParameters shouldUseBroker])
-    {
-        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorBrokerNotAvailable, @"Broker is not enabled for this operation. Please make sure you have enabled broker support for your application", nil, nil, nil, nil, nil, YES);
-        completionBlock(nil, error);
-        return;
-    }
-    
-    
-    if (![MSIDSSOExtensionGetDeviceInfoRequest canPerformRequest])
-    {
-        MSID_LOG_WITH_CTX(MSIDLogLevelInfo, requestParameters, @"Broker is not present on this device. Defaulting to personal mode");
-        
-        MSALDeviceInformation *msalDeviceInfo = [MSALDeviceInformation new];
-        msalDeviceInfo.deviceMode = MSALDeviceModeDefault;
-        completionBlock(msalDeviceInfo, nil);
-        return;
-    }
-    
-    NSError *requestError;
-    MSIDSSOExtensionGetDeviceInfoRequest *ssoExtensionRequest = [[MSIDSSOExtensionGetDeviceInfoRequest alloc] initWithRequestParameters:requestParameters
-                                                                                                                                  error:&requestError];
-    
-    if (!ssoExtensionRequest)
-    {
-        completionBlock(nil, requestError);
-        return;
-    }
-    
-    if (![self setCurrentSSOExtensionRequest:ssoExtensionRequest])
+    MSALDeviceInformation *deviceInformation = [MSALDeviceInformation new];
+
+    MSIDRegistrationInformation* regInfo = [MSIDWorkPlaceJoinUtil getRegistrationInformation:nil urlChallenge:nil];
```

Comment:
Could this code be inside MSALDeviceInformation helper method perhaps? That will help you to minimize changes in existing MSAL logic.

Reply:
If I understood code flow correctly then MSALPublicClientApplication providing interface to get MSALDeviceInformation object.
Here MSALDeviceInformation class only support setter and getter(internal header) methods and we don't have any logic in that to fetch data but its MSALDeviceInfoProvider who responsible for initialising MSALDeviceInformation.

So MSALPublicClientApplication takes help from MSALDeviceInfoProvider to get MSALDeviceInformation.
Any thoughts?